numpy/_core/umath.py

```diff
@@ -0,0 +1,33 @@
+"""
+Create the numpy._core.umath namespace for backward compatibility. In v1.16
+the multiarray and umath c-extension modules were merged into a single
+_multiarray_umath extension module. So we replicate the old namespace
+by importing from the extension module.
```

Comment:
I wonder if we need this module in `numpy._core` - is numpy internally still using these imports? I guess it would make the renaming a bit more complicated if downstream is using things from this namespace on purpose, but also almost all of this stuff is available in the main namespace.

Reply:
When I search for `_core.umath` in the codebase it's used/imported in a number of places (especially in `code_generators`). I see that there are also downstream usages but this PR provides shim for `numpy.core.umath` so downstream will still work (with a deprecation warning).
My guess would be to leave this module.

Reply:
SGTM